dev/tools/gen_keycodes/README.md

```diff
@@ -58,109 +58,62 @@ the API.
+  ArrowUp (0x304), F1 (0x801), Hiragata (0x716), and TVPower (0xD4B).
+  Some keys in the Chromium key list are not present in this plane, most notably
+  modifiers keys (such as Shift).  See the Flutter plane below for more
+  information.
+
+- **Plane 0x02**: The Flutter plane. This plane contains keys that are
+  defined by Flutter.  The values are also manually assigned by Flutter.
+  Modifier keys are placed in this plane, because Flutter distinguishes
+  between sided modifier keys (for example "ShiftLeft" and "ShiftRight"),
+  while Web doesn't (only has "Shift").
+  Other examples are numpad keys and gamepad keys.
+
+- **Plane 0x03-0x0F**: Reserved.
+
+- **Plane 0x10-0x1F**: Platform planes managed by Flutter.  Each platform plane
+  corresponds to a Flutter embedding officially supported by Flutter.  The
+  platforms are listed as follows:
 
   | Code | Platform |
   | ---- | -------- |
-  | 0x02 | Android  |
-  | 0x03 | Fuchsia  |
-  | 0x04 | iOS      |
-  | 0x05 | macOS    |
-  | 0x06 | Linux    |
-  | 0x07 | Windows  |
-  | 0x08 | Web      |
-  | 0xFF | Custom   |
-
-  Further ranges will be added as platforms are added. The platform prefix
-  does not define the platform it is used on, it is just the platform that
-  decides what the value is: the codes are mapped to the same value on all
-  platforms.
-
-- **0x100 0000 0000 - 0x1FF FFFF FFFF**: For keys that have no definition yet in
-  Flutter, but that are encountered in the field, this range is used to embed
-  the platform-specific keycode in an ID that must be tested for in a
-  platform-specific way. For instance, if a platform generates a new USB
-  HID code 0x07 00E8 that a Flutter app wasn’t compiled with, then it would
-  appear in the app as 0x100 0007 00E8, and the app could test against that
-  code. Yes, this also means that once they recompile with a version of
-  Flutter that supports this new HID code, apps looking for this code will
-  break. This situation is only meant to provide a fallback ability for apps
-  to handle esoteric codes that their version of Flutter doesn’t support yet.
-  The prefix for this code is the platform prefix from the previous sections,
-  plus 0x100.
-
-- **0x200 0000 0000 - 0x2FF FFFF FFFF**: For pseudo-keys which represent
-  combinations of other keys, and conceptual keys which don't have a physical
-  representation. This is where things like key synonyms are defined (e.g.
-  "shiftLeft" is a synonym for "shift": the "shift" key is a pseudo-key
-  representing either the left or right shift key).
-
-**This is intended to get us out of the business of defining key codes where
-possible.** We still have to have mapping tables, but at least the actual minting
-of codes is deferred to other organizations to a large extent. Coming up with a
-code is a mechanical process consisting of just picking the lowest number code
-possible that matches the semantic meaning of the key according to the
-definitions above.
-
-Here are some examples:
```

Comment:
It doesn't need to stay in this README, but I think it would be good to add some examples to explain the expected events, whether it is here or in `HardwareKeyboard` or another README -- someplace that someone maintaining this code will be likely to see it. If it is in a separate design doc that isn't part of the repo, I'm pretty sure that it will get out of date and/or overlooked.

Reply:
Makes sense, I'll add it to `HardwareKeyboard` later.